XenModel/Actions/AsyncAction.cs

```diff
@@ -366,9 +367,15 @@ private bool Poll(double start, double finish)
                     log.InfoFormat("Task {0} finished successfully", RelatedTask.opaque_ref);
                     if (task.result != "") // Work around CA-6597
                     {
-                        Match m = Regex.Match(task.result, "<value>(.*)</value>");
-                        if (m.Success)
-                            Result = m.Groups[1].Value;
+                        var doc = new XmlDocument();
+                        doc.LoadXml(task.result);
+                        var nodes = doc.GetElementsByTagName("value");
+
+                        foreach (XmlNode node in nodes)
+                        {
+                            Result = node.InnerText;

```

Comment:
Is the written `Result` even used for anything, it seems to be a local variable that doesn't get outputted anywhere?
Perhaps I've missed something with how async tasks work though.